src/pages/Account/NewCardForm.tsx

```diff
@@ -0,0 +1,278 @@
+      },
+      name: event.currentTarget.value,
+    })
+  }
+
+  handleStripeChange = (event: any): void => {
+    const { elementType, error } = event;
+    const errorMessage = error ? error.message : '';
+    return this.setState({
+      completed: {
+        ...this.state.completed,
+        [elementType]: event.complete,
+      },
+      errors: {
+        ...this.state.errors,
+        [elementType]: errorMessage,
+        submit: '',
+      },
+    });
+  };
+  
+  isCompleted = () => {
+    const { completed } = this.state;
+    for (let key in completed) {
+      if (!completed[key]) return false;
+    }
+    return true;
+  }
+
+  hasErrors = () => {
+    const { errors } = this.state;
+    for (let key in errors) {
+      if (errors[key]) return true;
+    }
+    return false;
+  }
+
+  handleSubmit = (event: React.FormEvent): void => {
+    event.preventDefault();
+    const { createPaymentSource, handleModal,  setAlert, stripe } = this.props;
+
+    this.setState({ isSubmitting: true },
+      () => (
+        stripe.createToken({ name: this.state.name })
+          .then(({ token }: any) => {
+            return createPaymentSource(token.id);
+          })
+          .then(() => {
+            this.setState({ isSubmitting: false }, handleModal);
+            setAlert({
+              color: 'success',
+              msg: 'Success! Your new card has been added.',
+              show: true,
+            });
+          })
+          .catch((error: Error) => {
+            console.error('error: ', error);
+            this.setState({
+              isSubmitting: false,
+              errors: {
+                ...this.state.errors,
+                submit: error.message,
+              }
+            });
+          })
+      )
+    );
+  };
+};
+
+const EnhancedNewCardForm = compose(
+  graphql(CREATE_PAYMENT_SOURCE, {
+    props: ({ mutate }: any) => ({
+      createPaymentSource: (stripeToken: string): Promise<any> => {
+        return mutate({
+          variables: { stripeToken },
+          refetchQueries: [{ query: GET_PAYMENT_SOURCES }],
+          update: (store: any, { data }: any) => {
+            if (!store.data.data.ROOT_QUERY || !store.data.data.ROOT_QUERY.getPaymentSources) {
+              return;
+            }
+
+            const { createPaymentSource } = data;
+            const paymentSources = store.readQuery({ query: GET_PAYMENT_SOURCES });
+            const { getPaymentSources } = paymentSources;
+            store.writeQuery({ query: GET_PAYMENT_SOURCES,
+              data: {
+                getPaymentSources: [
+                  ...getPaymentSources,
+                  createPaymentSource
+                ]
+              }
+            });
+          },
+        });
+      },
+    }),
+  })
+)(injectStripe(NewCardForm));
+
+const StripeWrappedNewCardForm = ({ handleModal, setAlert }: any) => (
+  <StripeWrapper>
+    <EnhancedNewCardForm handleModal={handleModal} setAlert={setAlert} />
```

Comment:
can you clarify what `handleModal` does? is handleModalOpen? or doing something else

Reply:
`handleModal` sets the modal for the page. It takes 2 params: ModalType enum and a sets the current paymentSource if necessary (for the DeleteCard). It can control which type of modal to open as well as close modals if no arg is given.

```
function handleModal(modal?: ModalType, newPaymentSource?: PaymentSource) {
    setModal(modal)
    newPaymentSource && setPaymentSource(newPaymentSource); // null check prevents modal from rendering undefined paymentSource when fading out
  }
```

Reply:
maybe `handleModalAction` ?

Reply:
sg